src/Action.php

```diff
@@ -3,68 +3,86 @@
 namespace Kirschbaum\Actions;
 
 use Throwable;
-use Illuminate\Support\Arr;
 use Kirschbaum\Actions\Contracts\Actionable;
+use Kirschbaum\Actions\Exceptions\ActionableInterfaceNotFoundException;
 
 class Action
 {
+    /**
+     * Arguments to pass into the action's constructor.
+     *
+     * @var array
+     */
+    protected $arguments;
+
     /**
      * Initiate the given action.
      *
-     * @param Actionable $action
+     * @param string $action
      *
      * @throws Throwable
      *
      * @return mixed|void
      */
-    public function act(Actionable $action)
+    public function act(string $action)
     {
+        $this->arguments = array_slice(func_get_args(), 1);
```

Comment:
Could we use the `...` syntax here (and the other actX methods)? Could be handy for LSPs that show what parameters can be passed into a method.

For example:
```suggestion
    public function act(string $action, ...$arguments)
    {
        $this->arguments = $arguments;
```

Reply:
You are absolutely right